lib/Mojolicious/Guides/Growing.pod

```diff
@@ -0,0 +1,776 @@
+              |                |     +-------+
+              |   Controller   |
+              |                |     +-------+
+  Response <- |                | <-> | View  |
+              +----------------+     +-------+
+
+The I<controller> receives a request from a user, passes incoming data to the I<model> and retrieves data from it,
+which then gets turned into an actual response by the I<view>. But note that this pattern is just a guideline that most
+of the time results in cleaner more maintainable code, not a rule that should be followed at all costs.
+
+=head2 REpresentational State Transfer
+
+REST is a software architectural style for distributed hypermedia systems such as the web. While it can be applied to
+many protocols it is most commonly used with HTTP these days. In REST terms, when you are opening a URL like
+C<http://mojolicious.org/foo> with your browser, you are basically asking the web server for the HTML I<representation>
+of the C<http://mojolicious.org/foo> I<resource>.
+
+  +--------+                                  +--------+
+  |        | -> http://mojolicious.org/foo -> |        |
+  | Client |                                  | Server |
+  |        | <-  <html>Mojo rocks!</html>  <- |        |
+  +--------+                                  +--------+
+
+The fundamental idea here is that all resources are uniquely addressable with URLs and every resource can have
+different representations such as HTML, RSS or JSON. User interface concerns are separated from data storage concerns
+and all session state is kept client-side.
+
+  +---------+                        +------------+
+  |         | ->    PUT /foo      -> |            |
+  |         | ->    Hello World!  -> |            |
+  |         |                        |            |
+  |         | <-    201 CREATED   <- |            |
+  |         |                        |            |
+  |         | ->    GET /foo      -> |            |
+  | Browser |                        | Web Server |
+  |         | <-    200 OK        <- |            |
+  |         | <-    Hello World!  <- |            |
+  |         |                        |            |
+  |         | ->    DELETE /foo   -> |            |
+  |         |                        |            |
+  |         | <-    200 OK        <- |            |
+  +---------+                        +------------+
+
+While HTTP methods such as C<PUT>, C<GET> and C<DELETE> are not directly part of REST they go well with it and are
+commonly used to manipulate I<resources>.
+
+=head2 Sessions
+
+HTTP was designed as a stateless protocol, web servers don't know anything about previous requests, which makes
+user-friendly login systems tricky. Sessions solve this problem by allowing web applications to keep stateful
+information across several HTTP requests.
+
+  GET /login?user=sebastian&pass=s3cret HTTP/1.1
+  Host: mojolicious.org
+
+  HTTP/1.1 200 OK
+  Set-Cookie: sessionid=987654321
```

Comment:
Ah, perfect example that not _all_ session state is kept client side.  In this case, "sebastian" is kept server-side and looked up from the client-side sessionid.